### PR TITLE
Fixing - IDENTITY-4109

### PR DESCRIPTION
**Pull request by JKAUSHALYA**
https://wso2.org/jira/browse/IDENTITY-4109

### CODE REVIEW

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/handlers/TokenResponseTypeHandler.java [hunk 1]
```diff
@@ -168,18 +166,16 @@ public OAuth2AuthorizeRespDTO issue(OAuthAuthzReqMessageContext oauthAuthzMsgCtx
                     // token is active and valid
                     if (log.isDebugEnabled()) {
                         if(expiryTime > 0){
```
Comment (omindu) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/handlers/TokenResponseTypeHandler.java]: Formatting issue.

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/RefreshGrantHandler.java [hunk 2]
```diff
@@ -107,8 +106,7 @@ public boolean validateGrant(OAuthTokenReqMessageContext tokReqMsgCtx)
             return false;
         }else if(!refreshToken.equals(accessTokenDO.getRefreshToken())){
```
Comment (omindu) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/RefreshGrantHandler.java]: Formatting issue.

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java [hunk 3]
```diff
@@ -556,7 +556,7 @@ public static long getRefreshTokenExpireTimeMillis(AccessTokenDO accessTokenDO)
         long refreshTokenValidityPeriodMillis = accessTokenDO.getRefreshTokenValidityPeriodInMillis();
 
         if (refreshTokenValidityPeriodMillis < 0) {
-            log.debug("Refresh Token : " + accessTokenDO.getRefreshToken() + " has infinite lifetime");
+            log.debug("Refresh Token has infinite lifetime");
```
Comment (omindu) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java]: check ```isDebugEnabled()```.

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java [hunk 4]
```diff
@@ -579,7 +579,7 @@ public static long getAccessTokenExpireMillis(AccessTokenDO accessTokenDO) {
         long validityPeriodMillis = accessTokenDO.getValidityPeriodInMillis();
 
         if (validityPeriodMillis < 0) {
-            log.debug("Access Token : " + accessTokenDO.getAccessToken() + " has infinite lifetime");
+            log.debug("Access Token has infinite lifetime");
```
Comment (omindu) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java]: check ```isDebugEnabled()```.

Comment (JKAUSHALYA) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java]: Fixed in https://github.com/wso2/carbon-identity/pull/1690

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java (hunk 3 above)
Comment (JKAUSHALYA) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java]: Fixed in https://github.com/wso2/carbon-identity/pull/1690

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/RefreshGrantHandler.java (hunk 2 above)
Comment (JKAUSHALYA) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/RefreshGrantHandler.java]: Fixed in https://github.com/wso2/carbon-identity/pull/1690

File: components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/handlers/TokenResponseTypeHandler.java (hunk 1 above)
Comment (JKAUSHALYA) [inline components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/handlers/TokenResponseTypeHandler.java]: Fixed in https://github.com/wso2/carbon-identity/pull/1690

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityLogTokenParser.java [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+            fileInput = new FileInputStream(file);
+
+            Properties properties = new Properties();
+            properties.load(fileInput);
+
+            for (String propertyName : properties.stringPropertyNames()) {
+                logTokenMap.put(propertyName, properties.getProperty(propertyName));
+            }
+        } catch (IOException e) {
+            log.error("An error occur while reading the file", e);
```
Comment (lalaji) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityLogTokenParser.java]: Here,is it sufficient only to log the error? Don't we need to throw it.

Comment (JKAUSHALYA) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityLogTokenParser.java]: We do not need to break the flow in here on a Exception and this is the top most level.

File: components/application-authenticators/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java [hunk 6]
```diff
@@ -647,7 +646,7 @@ protected String sendRequest(String url, String accessToken)
 
         reader.close();
```
Comment (msmshariq) [inline components/application-authenticators/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java]: It's best to do clean up operations in a ```finally``` block. If an exception is thrown, clean up won't happen unless it's done in the ```finally``` block.

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityLogTokenParser.java [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+     * Get the properties as a map.
+     * @return Map{String String}
+     */
+    Map<String, String> getLogTokenMap() {
+
+        return logTokenMap;
+    }
+
+    private static void buildConfiguration() {
+
```
Comment (chrishantha) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityLogTokenParser.java]: Why blank line? Check other methods also

File: components/application-authenticators/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java (hunk 6 above)
Comment (JKAUSHALYA) [inline components/application-authenticators/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java]: Fixed in https://github.com/wso2/carbon-identity/pull/1701

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityLogTokenParser.java (hunk 7 above)
Comment (JKAUSHALYA) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityLogTokenParser.java]: Fixed in https://github.com/wso2/carbon-identity/pull/1701

**Pull request merged**